test/client/fixtures/sandbox/node/classes-test.js

```diff
@@ -56,6 +56,59 @@ if (window.EventSource) {
 
         ok(eventSource instanceof EventSource);
     });
+
+    test('should have static constants (GH-1106)', function () {
+        ok(EventSource.hasOwnProperty('CONNECTING'));
+        ok(EventSource.hasOwnProperty('OPEN'));
+        ok(EventSource.hasOwnProperty('CLOSED'));
+    });
+
+    test('should process url to proxy with special flag (GH-1106)', function () {
```

Comment:
Add [CORS handling](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/request-pipeline/index.js#L50) and replace proxied url for `origin` property for EventSource requests.
Example:
```js
var eventSource= new EventSource("http://cross-domain.com", {
  withCredentials: true
});
eventSource.addEventListener('message', function(e) {
  if (e.origin === 'http://origin-site-url') ...;
});
```

Also let's add more real life tests:
- `EventSource` for same domain request.
- `EventSource` for cross domain request. 
- checking `origin` property